gta/map.py

```diff
@@ -14,120 +14,112 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-# Initially ported from OpenRW
-# https://github.com/rwengine/openrw/blob/master/rwengine/src/loaders/LoaderIPL.cpp
-
 from enum import Enum
+import os.path
 from collections import namedtuple
+from map_structures import III_structures, VC_structures, SA_structures
 
 # Data types
-# IPL and IDE section formats taken from
-# gta.fandom.com/wiki/Item_Placement
-# gta.fandom.com/wiki/IDE
-
 Vector = namedtuple("Vector", "x y z")
 
-#######################################################
-class IPL:
+# List of IPL/IDE sections which require a section utility that's different
+# from the default one. E.g. sections with variable number of parameters in
+# their data structures
+specialSections = [
+    'objs': OBJSSectionUtility,
+    'tobj': TOBJSectionUtility
+]
 
-    sections = {
-        'inst': IPLSection_INST,
-        'pick': IPLSection_PICK,
-        'cull': IPLSection_CULL,
-        'zone': IPLSection_ZONE
-    }
+# Utility for reading / writing to map data files (.IPL, .IDE)
+# Returns a dictionary of sections found in the given file
+#######################################################
+class MapDataUtility:
 
     #######################################################
-    def __init__(self):	
-        self.inst_list = []
-        self.pickup_list = []
-        self.cull_list = []
-        self.zone_list = []
+    def read(filename, dataStructures):
 
-    #######################################################
-    def read(self, filename):
+        if not os.path.exists(filename):
+            print("MapDataUtility error: File not found")
```

Comment:
Would prefer if it threw an exception instead.

Reply:
Ok sure!